src/cogent3/parse/genbank.py

```diff
@@ -348,85 +378,135 @@ def __str__(self):
-                if self.Ambiguity:
-                    curr = self.Ambiguity + str(data)
+                if self.ambiguity:
+                    curr = self.ambiguity + str(data)
                 else:
                     curr = str(data)
             except TypeError:
                 # if long conversion failed, should have two LocalLocation
                 # objects
                 first, last = self._data
-                if self.IsBounds:
+                if self.is_bounds:
                     curr = f"({first}{'.'}{last})"
                 else:
                     curr = f"{first}{'..'}{last}"
         # check if we need to add on the accession and database
-        if self.Accession:
-            curr = self.Accession + ":" + curr
-            # we're only going to add the Db if we got an accession
-            if self.Db:
-                curr = self.Db + "::" + curr
+        if self.accession:
+            curr = self.accession + ":" + curr
+            # we're only going to add the db if we got an accession
+            if self.db:
+                curr = self.db + "::" + curr
         # check if it's complemented
-        if self.Strand == -1:
+        if self.strand == -1:
             curr = f"complement({curr})"
         return curr
 
-    def first(self):
+    @property
+    def start(self):
         """Returns first base self could be."""
         try:
-            return int(self._data)
+            return int(self._data) - 1
         except TypeError:
-            return self._data[0].first()
+            return self._data[0].start
 
-    def last(self):
+    @property
+    def stop(self):
         """Returns last base self could be."""
         try:
-            return int(self._data)
+            return int(self._data) - 1
         except TypeError:
-            return self._data[-1].last()
+            return self._data[-1].stop
 
+    def first(self):
+        from cogent3.util.warning import deprecated
+
+        deprecated(
+            "method",
+            "Location.first",
+            "Location.start",
+            "2023.06.15",
+            "Warning: '.start' is changed to reflect 0-based coordinated, previous implementation reflected 1-based coordinates",
+        )
 
-class LocationList(list):
-    """List of Location objects.
+        return self.start
 
-    WARNING: Coordinates are based on 1, not 0, to match GenBank format.
-    """
+    def last(self):
+        from cogent3.util.warning import deprecated
+
+        deprecated(
+            "method",
+            "Location.last",
+            "Location.stop",
+            "2023.06.15",
+            "Warning: '.stop' is changed to reflect 0-based coordinated, previous implementation reflected 1-based coordinates",
+        )
+        return self.stop
+
+
+class LocationList(list):
+    """List of Location objects."""
 
     BIGNUM = 1e300
 
-    def first(self):
+    def start(self):
         """Returns first base of self."""
         curr = self.BIGNUM
```

Comment:
These methods are rather old. The `start()` is just finding the minimum value, so the `BIGNUM` attribute can be removed entirely and the method rewritten as
```python
    def start(self):
        """Returns first base of self."""
        return min(self, key=lambda x: x.start)

    def stop(self):
        """Returns last base of self."""
        return max(self, key=lambda x: x.stop)
```

I note that the methods on `LocationList` don't seem to be tested, can you just add tests -- check that the original code works and inspect the output being generated under it. Make sure it makes sense!

Then change to using your new properties and updater the tests to handle the 0-based counting.

Reply:
In looking over these again I can see the `first()` and `last()` methods on `LocationList` are never used, so they should be discontinued. No tests needed!

Reply:
Easy!